packages/console/src/monaco/MonacoProviders.tsx

```diff
@@ -14,13 +16,119 @@ interface MonacoProviderProps {
   language: string;
 }
 
+const DEFAULT_RUFF_SETTINGS = {
+  preview: true,
+  'target-version': 'py38',
+  'line-length': 88,
+  'indent-width': 4,
+  format: {
+    'indent-style': 'space',
+    'quote-style': 'double',
+  },
+  lint: {
+    'flake8-implicit-str-concat': {
+      'allow-multiline': false,
+    },
+    ignore: ['ISC003'],
+    select: [
+      'F',
+      'E1',
+      'E9',
+      'E711',
+      'W291',
+      'W293',
+      'W605',
+      'I002',
+      'B',
+      'A',
+      'COM818',
+      'ISC',
+      'PLE',
+      'RUF001',
+      'RUF021',
+      'RUF027',
+      'PLR1704',
+      'LOG',
+      'ASYNC',
+      'RET501',
+      'RET502',
+      'RET503',
+      'PLC2401',
+      'PLC2403',
+      'NPY',
+      'PERF',
+      'C4',
+    ],
+  },
+};
+
 /**
  * Registers a completion provider with monaco for the language and session provided.
  */
 class MonacoProviders extends PureComponent<
   MonacoProviderProps,
   Record<string, never>
 > {
+  static workspace?: Workspace;
+
+  static initRuffPromise?: Promise<void>;
+
+  /**
+   * Loads and initializes Ruff.
+   * Subsequent calls will return the same promise.
+   */
+  static async initRuff(): Promise<void> {
+    if (MonacoProviders.initRuffPromise) {
+      return MonacoProviders.initRuffPromise;
+    }
+
+    log.debug('Initializing Ruff');
+
+    MonacoProviders.initRuffPromise = init().then(() => {
+      MonacoProviders.setRuffSettings();
+    });
+
+    return MonacoProviders.initRuffPromise;
+  }
+
+  static async setRuffSettings(
+    settings: Record<string, unknown> = DEFAULT_RUFF_SETTINGS
+  ): Promise<void> {
+    await MonacoProviders.initRuff();
+
+    MonacoProviders.workspace = new Workspace(settings);
+
+    monaco.editor
+      .getModels()
+      .filter(m => m.getLanguageId() === 'python')
+      .forEach(MonacoProviders.lintPython);
+  }
+
+  static lintPython(model: monaco.editor.ITextModel): void {
+    if (!MonacoProviders.workspace) {
+      return;
+    }
+
+    monaco.editor.setModelMarkers(
+      model,
+      'ruff',
+      MonacoProviders.workspace.check(model.getValue()).map((d: Diagnostic) => {
+        const isUnnecessary = d.code === 'F401' || d.code === 'F841';
```

Comment:
This is unused variable and unused import. It is mainly to add the unnecessary tag in monaco which makes the text appear grayed-out. I took this from ruff's playground code

Ruff doesn't distinguish between error and warning diagnostics (there is pycodestyle error and warning, but that's just the rule names). So it would be up to us to determine if we wanted configuration for more as warnings instead of errors

Reply:
Okay, was just expecting implicitly concatenated strings to be a warning not an error.